views/overlays/matchIntro.hbs

```diff
@@ -29,156 +33,169 @@
-      document.getElementById("audio-chime").pause();
-
-      var initialBackground = 'url(http://i.imgur.com/tlhS4vA.png)';
-      var step = 0;
       var socket = io('/overlay');
-
-      socket.on('play intro', function(data) {
-        if(data.flag !== step) { 
-          return true;
-        }
-
-        var showArr = [$('#tourneyInfo'), $('#lplayer'), $('#rplayer')];
-
-        if (data.flag === 0) {
-          // Reset the page and fade out background
-          $('body').css('background-color', 'rgba(0,0,0,0)');
-          $('#container').animate({'opacity': '0'}, 500, function () {
-              $('#container').css('background', initialBackground);
-              $('#lCharacter').css({'opacity': '0', 'margin-left': '-3%'});
-              $('#rCharacter').css({'opacity': '0', 'margin-left': '-3%'});     
-              $('#vs').css('font-size', '100em').hide();
-              showArr.forEach(function(elem) {
-                elem.hide();
-              });
-            });
-
-            step = 1;
-        } else if (data.flag === 1) {
-          // Update the page with all relevant data/images
-          $('#tourneyInfo').text(data.tourneyInfo || '');
-          $('#lplayer').text(data.lplayer || '');
-          $('#rplayer').text(data.rplayer || '');
-          $('#lCharacter').attr('src', charImageDict[data.lCharacter] || '');
-          $('#rCharacter').attr('src', charImageDict[data.rCharacter] || '');
-
-          step = 2;
-        } else {
-          // Show the background again and begin animation
-          $('#container').animate({'opacity': '1'}, 500, function () {
-            $('body').css('background-color', 'rgba(255,255,254,1)');
-            setTimeout(function() {
-              $('#lCharacter').trigger('fadeIn');
-              setTimeout(function () {
-                $('#rCharacter').trigger('fadeIn');
-                setTimeout(function () {
-                  transitionBackground(showArr);
-                }, 500); 
-              }, 500); 
-            }, 100);  
-          }); 
-
-          step = 0;            
-        }     
-      });
       
-      $('.fadeIn').bind('fadeIn', function() {
-          $(this).animate({'opacity': '1', 'margin-left': '0'}, 350);
+      // For some reason the autoplay property doesn't work in Chrome
+      document.getElementById("audio-chime").pause();   
+      CreateMainTimeline();
+      
+      socket.on('update overlay', function(data) {
+        $('#tourneyInfo').text(data.tourneyInfo || '');
+        $('#lplayer').text(data.lplayer || '');
+        $('#rplayer').text(data.rplayer || '');
+        $('#lCharacter').attr('src', charImageDict[data.lCharacter] || '');
+        $('#rCharacter').attr('src', charImageDict[data.rCharacter] || '');
       });
+
+      socket.on('play intro', function(label) {
+        if (!inProgress) {
+          inProgress = true;
+          PlayLabel(label);
+        }  
+      });      
     });
+    
+    function CreateMainTimeline() {
```

Comment:
Javascript function names traditionally use `camelCase`. C# and the like use `PascalCase`. I'm not sure if @dguenther cares either way, but this is my only gripe.